packages/synapse-react-client/src/components/Plot/SynapsePlot.tsx

```diff
@@ -1,135 +1,164 @@
-      partMask: SynapseConstants.BUNDLE_MASK_QUERY_RESULTS,
-      entityId: parseEntityIdFromSqlStatement(query),
-      query: {
-        sql: query,
-      },
-    }
-
-    getFullQueryTableResults(queryRequest, this.context.accessToken)
-      .then((data: QueryResultBundle) => {
-        this.setState({
-          isLoaded: true,
-          queryData: data,
-        })
-      })
-      .catch((err: any) => {
-        console.log('Error on full table query ', err)
-      })
+  if (!queryData) {
+    return <></>
   }
 
-  public showPlot() {
-    if (!this.state.isLoaded) {
-      return
-    }
-    const { title, xtitle, ytitle, type, xaxistype, showlegend } =
-      this.props.widgetparamsMapped
-    const queryData = this.state
-    const isHorizontal = this.props.widgetparamsMapped.horizontal.toLowerCase()
-    const layout: any = {
-      showlegend,
-      title,
-    }
-    if (xtitle) {
-      layout.xaxis = {
-        title: xtitle,
-      }
+  const {
+    title,
+    xtitle,
+    ytitle,
+    type,
+    xaxistype,
+    showlegend,
+    barmode,
+    horizontal,
+    displayModebar,
+  } = props.widgetparamsMapped
+  const isShowLegend = toBoolean(showlegend)
+  const isHorizontal = toBoolean(horizontal)
+  const isDisplayModebar = toBoolean(displayModebar)
+  const config: Partial<Plotly.Config> = {
+    displayModeBar: isDisplayModebar,
+  }
+  const layout: Partial<Plotly.Layout> = {
+    showlegend: isShowLegend,
+    title,
+    barmode: barmode
+      ? (barmode.toLowerCase() as 'stack' | 'group' | 'overlay' | 'relative')
+      : undefined,
+  }
+  if (xtitle) {
+    layout.xaxis = {
+      title: xtitle,
     }
-    if (xaxistype) {
-      layout.xaxis = {
-        ...layout.xaxis,
-        xaxistype: xaxistype.toLowerCase(),
-      }
+  }
+  if (xaxistype) {
+    layout.xaxis = {
+      ...layout.xaxis,
+      type: xaxistype.toLowerCase() as AxisType,
     }
-    if (ytitle) {
-      layout.yaxis = {
-        title: ytitle,
-      }
+  }
+  if (ytitle) {
+    layout.yaxis = {
+      title: ytitle,
     }
-    // init plot_data
-    const plotData: any = []
-    const orientation = isHorizontal ? 'v' : 'h'
-    const headers = queryData.queryData.queryResult?.queryResults.headers ?? []
-    for (let i = 0; i < headers.length - 1; i += 1) {
-      // make an entry for each set of data points
-      plotData[i] = {
-        orientation,
-        name: headers[i + 1].name,
-        type: type.toLowerCase(),
-        x: [],
-        y: [],
-      }
+  }
+  // init plot_data
+  const plotData: Partial<Plotly.PlotData>[] = []
+  const orientation = isHorizontal ? 'h' : 'v'
+  const headers = queryData.queryResult?.queryResults.headers ?? []
+  for (let i = 0; i < headers.length - 1; i += 1) {
+    // make an entry for each set of data points
+    plotData[i] = {
+      orientation,
+      name: headers[i + 1].name,
+      type: type.toLowerCase() as PlotType,
+      x: [],
+      y: [],
+      customdata: [],
```

Comment:
Added customdata for each point, to pass Row back to click handler (if specified)